agent/acs/handler/acs_handler.go

```diff
@@ -223,15 +228,45 @@ func (acsSession *session) Start() error {
 					seelog.Debugf("Failed to write to deregister container instance event stream, err: %v", err)
 				}
 			}
+
 			if shouldReconnectWithoutBackoff(acsError) {
 				// If ACS closed the connection, there's no need to backoff,
 				// reconnect immediately
 				seelog.Infof("ACS Websocket connection closed for a valid reason: %v", acsError)
 				acsSession.backoff.Reset()
 				sendEmptyMessageOnChannel(connectToACS)
+				// Check if this instance has ECS_DISCONNECT_CAPABLE turned on. This code block is only reached when
+				// ACS has already failed to connect.
+			} else if cfg.DisconnectCapable.Enabled() {
+				if acsSession.disconnectionTimer != nil {
+					timerCompleted := acsSession.checkDisconnectionTimer()
+					logger.Debug("Checking disconnectionTimer status", logger.Fields{
+						"timerCompleted": timerCompleted,
+					})
+					// If the timer has been completed, then set DisconnectModeEnabled to true.
+					if timerCompleted {
+						cfg.SetDisconnectModeEnabled(true)
+						logger.Debug("Turning Disconnection Mode on after timer is completed", logger.Fields{
+							"disconnectionMode": cfg.GetDisconnectModeEnabled(),
+						})
+						// If the timer has not been completed, then attempt to connect to ACS every minute (to avoid
+						// excessive connection attempts).
+					} else {
+						logger.Debug("Starting 1 minute timer to reconnect to ACS")
+						intervalComplete := acsSession.waitForDuration(1 * time.Minute)
```

Comment:
nit. maybe define a constant `reconnectToACSTimeout` for `1 * time.Minute`, and use the new constant instead of hard-coded it in the debug log and intervalComplete.